core/txpool/legacypool/legacypool.go

```diff
@@ -1391,6 +1432,9 @@ func (pool *LegacyPool) runReorg(done chan struct{}, reset *txpoolResetRequest,
 	pool.changesSinceReorg = 0 // Reset change counter
 	pool.mu.Unlock()
 
+	// Transfer transactions from OverflowPool to MainPool for new block import
+	pool.transferTransactions()
```

Comment:
should call `transferTransactions()` at the beginning of `runReorg`?

Reply:
Premature/unnecessary transfers might happen especially in case the reorg is due to new block. Transferring transactions at the end avoids unnecessary processing of overflow transactions that might become invalid due to the new block.
Also the availability of space in the main pool is determined after the pending and queued transactions have been updated during the initial code(promotion, demotion, truncation). It ensures that the main pool is properly updated and cleaned before new transactions are brought in. 
So calling `transferTransactions()` at this point will ensure consistency, efficiency and new block adjustment better than calling it at the beginning I feel.